0034-connector-requirements/0034-connector-requirements.md

```diff
@@ -0,0 +1,150 @@
+
+When a node's routing data changes in such a way that it is necessary to notify its peers the node will broadcast the changes to all peers. Likewise, the node will continuously be processing incoming updates from its peers.
+
+## Special Addresses
+
+**`peer.*`**
+
+Routing data updates, IL-DCP and other peer-to-peer protocols use ILP packets where the destination address is in the `peer.*` address-space. Nodes that receive packets in the peer.* address-space MUST NOT forward these packets to another node.
+
+**`test.*`, `test1.*`, `test2.*`, and `test3.*`**
+
+A node MUST run either in a test network or on the live network but never on both. If a node is running on the test network it MUST reject all packets in the global address-space, `g.*`. Likewise, if node is running on the live netw    ork it MUST reject any packets with addresses in the `test.*`, `test1.*`, `test2.*`, or `test3.*` address-spaces.
+
+## Settlement
+
+When a node routes a packet it is accepting an offer, from the requesting peer, to pay for proof-of-delivery of that packet. When it returns a valid response to the requesting peer (an ILP Fulfill packet with the correct fulfillment) before the expiry of the request, this creates an obligation between the peers. The requesting peer now owes the forwarding peer the amount specified in the request packet.
+
+According to an agreed schedule the two peers will reconcile and settle the obligations created between them as a result of successfully forwarding ILP packets. It is important to note that this process is a bi-lateral concern and does not impact the settlement of obligations between other nodes involved in forwarding that packet.
+
+The specific schedule and mechansim for doing this will be specific to the settlement system used by the peers (e.g. payment channels on a distributed ledger, traditional wire transfers via the banking system, etc.), therefor the functionality required to do this is not included in the core node but rather in settlement-system-specific plugins or adaptors.
+
+It is necessary for the node to have a view of the current outstanding obligations with the peer (the peer's account balance) in order to apply appropriate risk management measures when processing packets from the peer.
+
+The logic that determines when to perform a settlement is currently implemented within the node (as opposed to in the plugin) and as such the node must instruct the plugin when to perform a settlement. This reduces the outstanding obligations with the peer to a limit that the node considers safe such that it can forward further packets for the peer.
```

Comment:
It seems that the entity that exactly determines when to settle (when to execute on-ledger transaction) is the plug-in (`_claimFunds` is not called externally). Is that ok?

Reply:
In this context `_claimFunds` is not settlement.

The sendMoney from the connector is what triggers settlement which is just the exchange of claims.